src/ecdsa/ecdh.py

```diff
@@ -109,57 +130,78 @@ def load_private_key(self, private_key):
 
     def load_private_key_bytes(self, private_key):
         """
-        Load private key from plain bytes string.
-        Uses current curve and checks if key length corresponds to
-        the current curve.
+        Load private key from byte string.
+        Uses current curve and checks if the provided key matches
+        the curve of ECDH key agreement.
+
         :param private_key: private key in bytes string format
-        :return: public (verifying) key from this private key
+        :type private_key: byte string
+
+        :raises NoCurveError: Curve must be set before loading.
+
+        :return: public (verifying) key from this private key.
+        :rtype: byte string
         """
+        if self.curve is None:
+            raise NoCurveError("Curve must be set prior to key load.")
         return self.load_private_key(
             SigningKey.from_string(private_key, curve=self.curve))
 
     def load_private_key_der(self, private_key):
         """
-        Load private key from DER string.
-        Curve is get from DER and check if it the same as current curve
+        Load private key from DER byte string.
+        Compares the curve of the DER-encoded key with the ECDH set curve,
+        uses the former if unset.
 
         Note, the only DER format supported is the RFC5915
         Look at keys.py:SigningKey.from_der()
 
         :param private_key: binary string with the DER encoding of private ECDSA key
-        :return: public (verifying) key from this private key
+        :type private_key:   string
```

Comment:
```
:type private_key: :term:`bytes-like object`
```

Reply:
doublechecked. pem and der loads via `string`